app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,61 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  skip_before_filter :verify_authenticity_token
+
+  def instagram
+    generic_callback( 'instagram' )
+  end
+
+  def facebook
+    generic_callback( 'facebook' )
+  end
+
+  def twitter
+    generic_callback( 'twitter' )
+  end
+
+  def google_oauth2
+    generic_callback( 'google_oauth2' )
+  end
+
+  def generic_callback( provider )
```

Comment:
Tried making these private, but it doesn't work due to devise accessing these for authentication